File: 219-contains-duplicate-ii/contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        my_set = set()

        for i in range(len(nums)):
            if i > k:
                my_set.remove(nums[i - k - 1])
            if nums[i] in my_set:
                return True
            my_set.add(nums[i])

        return False
    

File: 219-contains-duplicate-ii/contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    const hasmap = new Set();

    for (let i = 0; i < nums.length; i++) {
        if (i > k) {
            hasmap.delete(nums[i - k - 1]);
        }

        if (hasmap.has(nums[i])) {
            return true;
        }

        hasmap.add(nums[i]);
    }

    return false;
};
